source/Color.cpp

```diff
@@ -15,6 +15,17 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include "Color.h"
 
+#include "Preferences.h"
+
+#include <vector>
+
+namespace {
+	// Constants for each color filter mode to make for easier debugging.
```

Comment:
Just use an enum class, look in preferences how it is done. And if you are doing an enum also use a switch statement when checking which filter should get applied